CodeGeneration/Sources/SyntaxSupport/Traits.swift

```diff
@@ -110,21 +155,4 @@ public let TRAITS: [Trait] = [
       Child(name: "TrailingComma", kind: .token(choices: [.token(tokenKind: "CommaToken")]), isOptional: true)
     ]
   ),
-  Trait(
-    traitName: "MissingNode",
-    documentation: """
-      Represents a layout node that is missing in the source file.
-
-      See the types conforming to this protocol for examples of where missing nodes can occur.
-      """,
-    children: [
-      Child(
-        name: "Placeholder",
-        kind: .token(choices: [.token(tokenKind: "IdentifierToken")]),
-        documentation: """
-          A placeholder, i.e. `<#placeholder#>`, that can be inserted into the source code to represent the missing node.
-          """
-      )
-    ]
-  ),
```

Comment:
Speaking about that I've sent you a DM on swift.forums. I'll pass the message here to spark a discussion: 

I've noticed that we often try to arrange elements in alphabetical order in many areas. However, sometimes there is no clear instruction asking contributors to do so. Not to mention, it can be quite burdensome to both remember and actually perform this task. Have you considered automating it? Perhaps with swift-format?

It could work like this:

// swift-format: sorted_by:Node.getter:kind
public let DECL_NODES: [Node] = [
  Node(
    kind: .importPathComponent,
    base: .syntax,
    nameForDiagnostics: nil,
    children: (...)
  ),

  Node(
    kind: .importPath,
    base: .syntaxCollection,
    nameForDiagnostics: nil,
    elementChoices: [.importPathComponent]
  ),
(...)
]

Reply:
I have not really considered it because I believe that adding that adding this sort of tooling is non-trivial to implement since swift-format only has a syntactic view of the file and I wouldn’t know how best to describe by which property to sort the elements. For example, take the `visit` methods in `ParseDiagnsoticGenerator`. Some of them should be at the top and then from some point onwards they should be sorted by the type name of the first argument. 

So, my thinking is that building and maintaining this tooling would be a lot more effort than just sorting the elements manually. But if you’d be interested in building such tooling or have ideas of how to do it, I’d be very interested.